src/devices/QwiicButton/QwiicButton.QueueManipulation.cs

```diff
@@ -0,0 +1,119 @@
+﻿//// Licensed to the .NET Foundation under one or more agreements.
+//// The .NET Foundation licenses this file to you under the MIT license.
+//// See the LICENSE file in the project root for more information.
+
+using Iot.Device.QwiicButton.RegisterMapping;
+
+namespace Iot.Device.QwiicButton
+{
+    public sealed partial class QwiicButton
+    {
+        /// <summary>
+        /// Returns whether the queue of button press timestamps is full.
+        /// </summary>
+        public bool IsPressedQueueFull()
+        {
+            var pressedQueue = new QueueStatusBitField(_registerAccess.ReadSingleRegister(Register.PressedQueueStatus));
+            return pressedQueue.IsFull;
+        }
+
+        /// <summary>
+        /// Returns whether the queue of button press timestamps is empty.
+        /// </summary>
+        public bool IsPressedQueueEmpty()
+        {
+            var pressedQueue = new QueueStatusBitField(_registerAccess.ReadSingleRegister(Register.PressedQueueStatus));
+            return pressedQueue.IsEmpty;
+        }
+
+        /// <summary>
+        /// Returns how many milliseconds it has been since the last button press.
+        /// Since this returns a 32-bit unsigned int, it will roll over about every 50 days.
+        /// </summary>
+        public uint TimeSinceLastPress()
+        {
+            return _registerAccess.ReadQuadRegister(Register.PressedQueueFront);
+        }
+
+        /// <summary>
+        /// Returns how many milliseconds it has been since the first button press.
+        /// Since this returns a 32-bit unsigned int, it will roll over about every 50 days.
+        /// </summary>
+        public uint TimeSinceFirstPress()
```

Comment:
can you mention if any method will call this to reset? What will it return if it wasn't pressed at all?

Reply:
Method does not reset anything. Added documentation about what happens if queue is empty.